### ISSUE TITLE
shuffle should be set to True in train_loader of SinglePathTrainer

### ISSUE THREAD

**Issue by dutxubo** [microsoft/nni#4692]
**Brief description of your proposal if any**:  
in nni/nni/retiarii/oneshot/pytorch/random.py 
Line 144:
```python
self.train_loader = torch.utils.data.DataLoader(self.dataset_train,
batch_size=batch_size,
num_workers=workers)  
``` 
->
```python
self.train_loader = torch.utils.data.DataLoader(self.dataset_train,
batch_size=batch_size,
num_workers=workers,
shuffle=True)  
```

**Comment by matluster** [comment]
Excellent. Please open a pull request for this.
